lib/ecto/query/builder/join.ex

```diff
@@ -167,13 +168,18 @@ defmodule Ecto.Query.Builder.Join do
         %JoinExpr{qual: unquote(join_qual), source: unquote(join_source),
                   assoc: unquote(join_assoc), file: unquote(env.file),
                   line: unquote(env.line), params: unquote(join_params),
+                  hint: unquote(hint),
                   on: %QueryExpr{expr: unquote(on_expr), params: unquote(on_params),
                                  line: unquote(env.line), file: unquote(env.file)}}
       end
 
     Builder.apply_query(query, __MODULE__, [join], env)
   end
 
+  defmacro hint(expr) do
```

Comment:
Where is this macro used?

Reply:
To be clear, the formatting of the hint should not be done here in any level but exclusively on the adapter.